### ISSUE TITLE
FNX3-17102 ⁃ [Bug] Crash when trying to open multiple bookmarks or pages from history

### ISSUE THREAD

**Issue by softvision-miralobontiu** [mozilla-mobile/fenix#3553]
## Steps to reproduce
1. Make sure you have several bookmarks and history.
1. Go to Settings - Your Library - Bookmarks / History and select several pages.
1. From the overflow menu select "Open in private tabs".

### Expected behavior
The selected pages open in private tabs.

### Actual behavior
App crashes. Reproducible 1/3 times.

### Device information
* Android device: Google Pixel (Android Q)
* Fenix version: 1.0.0-rc.2 (#11700147)

[logcat.txt](https://github.com/mozilla-mobile/fenix/files/3306382/logcat.txt)

**Comment by vesta0** [comment]
@colintheshots thoughts?

**Comment by sblatz** [comment]
I'm not able to reproduce this after many attempts on my Galaxy S9. @softvision-miralobontiu are you starting in private mode or regular mode when going to the history menu?

**Comment by kbrosnan** [comment]
Looks like the bookmark may be a local file uri. Logcat shows `file:///storage/emulated/0/Download/unnamed.jpg`

```
06-19 12:07:57.799 20590 20590 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.android.systemui/com.android.systemui.chooser.ChooserActivity}: android.os.FileUriExposedException: file:///storage/emulated/0/Download/unnamed.jpg exposed beyond app through Intent.getData()
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3260)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3396)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2009)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7319)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:934)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: Caused by: android.os.FileUriExposedException: file:///storage/emulated/0/Download/unnamed.jpg exposed beyond app through Intent.getData()
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.os.StrictMode.onFileUriExposed(StrictMode.java:2083)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.net.Uri.checkFileUriExposed(Uri.java:2388)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.content.Intent.prepareToLeaveProcess(Intent.java:10746)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.content.Intent.prepareToLeaveProcess(Intent.java:10699)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.Instrumentation.execStartActivityAsCaller(Instrumentation.java:1959)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.Activity.startActivityAsCaller(Activity.java:5340)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at com.android.systemui.chooser.ChooserHelper.onChoose(ChooserHelper.java:43)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at com.android.systemui.chooser.ChooserActivity.onCreate(ChooserActivity.java:32)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7783)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7772)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1299)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3235)
06-19 12:07:57.799 20590 20590 E AndroidRuntime: 	... 11 more
```

**Comment by softvision-miralobontiu** [comment]
@sblatz I could reproduce the crash in normal browsing, and private browsing.

**Comment by ekager** [comment]
@softvision-miralobontiu is what @kbrosnan said above correct? Was one of your bookmark/history items a file URI?

**Comment by softvision-miralobontiu** [comment]
I'm not sure I had a URI file opened, but I could reproduce the crash again with Google Pixel (Android Q), Samsung Galaxy Note 8 (Android 9), Motorola Moto G6 (Android 8.1) on the latest build 1.0.1926 (#11761826). 
I selected only one bookmark/history page and chose "Open in private tabs" and the app crashed.
I only had Wikipedia pages opened and CNN articles.

The logcat from Google Pixel:
[logcat.txt](https://github.com/mozilla-mobile/fenix/files/3328413/logcat.txt)

The logcat from Motorola:
[logcat-Motorola.txt](https://github.com/mozilla-mobile/fenix/files/3328486/logcat-Motorola.txt)

**Comment by colintheshots** [comment]
Looking at the LogCats listed, the fix for #3785 should fix this crash as well. Please re-test after the next successful Nightly build.

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest build #11830608), with Google Pixel (Android Q), and Motorola Moto G6 (Android 8).
